harvester/actor.py

```diff
@@ -80,14 +130,140 @@ def has_person_data(self):
         """
         return self._person_dict is not None
 
+    @property
+    def _organization_uri(self):
+        """
+        Return URI for the affiliation of this actor.
+
+        The URIs are from http://uri.suomi.fi/codelist/fairdata/organization/code. For
+        FIN-CLARIAH affiliations, the home organization from the `departmentName` field
+        is used when determining the URI.
+
+        Organization names seem to always available in English, so English names are
+        used for matching.
+
+        Raises UnknownOrganizationException if URI match is not found.
+        """
+
+        organization_name = self.data["affiliation"]["organizationName_en"]
+
+        if organization_name == "FIN-CLARIN":
+            organization_name = self.data["affiliation"]["departmentName_en"]
+
+        url_base = "http://uri.suomi.fi/codelist/fairdata/organization/code"
+        organization_codes = {
+            "Aalto University": "10076",
+            "CSC — IT Center for Science Ltd": "09206320",
+            "Centre for Applied Language Studies": "01906-213060",
+            "National Library of Finland": "01901-H981",
+            "South Eastern Finland University of Applied Sciences": "10118",
+            "University of Eastern Finland": "10088",
+            "University of Helsinki": "01901",
+            "University of Jyväskylä": "01906",
+            "University of Oulu": "01904",
+            "University of Tampere": "10122",
+            "University of Turku": "10089",
+        }
+        if organization_name in organization_codes:
+            return f"{url_base}/{organization_codes[organization_name]}"
+        raise UnknownOrganizationException(
+            f"Could not determine URI for {organization_name}"
+        )
+
+    def _none_if_no_affiliation(func):
+        """
+        Wrapper for properties that are None if the actor does not have affiliation data
+        """
+        # Wrapper shouldn't have self despite being defined in class
+        # pylint: disable=no-self-argument
+
+        def wrapper(self, *args, **kwargs):
+            if "affiliation" not in self.data:
+                return None
+            # pylint: disable=not-callable
```

Comment:
Whaa, how is pylint inferring that `func` is not callable?

Reply:
No hecking idea. Might be that it thinks that the first argument would be `self` because we are in a class or something? Defining wrappers in a class feels a bit dirty after all. Just speculating though.